test/libsolidity/syntaxTests/controlFlow/storageReturn/assembly/switch_err.sol

```diff
@@ -0,0 +1,28 @@
+contract C {
+    struct S { bool f; }
+    S s;
+    function f(uint256 a) internal pure returns (S storage c) {
+        assembly {
+            switch a
+            case 0 { c_slot := s_slot }
+        }
+    }
+    function g(bool flag) internal pure returns (S storage c) {
+        assembly {
+            switch flag
+            case 0 { c_slot := s_slot }
+            case 1 { c_slot := s_slot }
+        }
+    }
+    function h(uint256 a) internal pure returns (S storage c) {
+        assembly {
+            switch a
+            case 0 { c_slot := s_slot }
+            default { return(0,0) }
+        }
+    }
+}
+// ----
+// TypeError: (96-107): This variable is of storage pointer type and can be returned without prior assignment, which would lead to undefined behaviour.
+// TypeError: (256-267): This variable is of storage pointer type and can be returned without prior assignment, which would lead to undefined behaviour.
+// TypeError: (459-470): This variable is of storage pointer type and can be returned without prior assignment, which would lead to undefined behaviour.
```

Comment:
I think the third should not be an error, as explained in the other comment: Whenever we reach the code that uses the Solidity variable `c` to return it, it has been assigned before. This is even the case for this being an external library function.